operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -379,7 +379,7 @@ func (c *cesMgr) getLargestAvailableCES() *cesTracker {
 		cepCount := len(ces.ces.Endpoints)
 		ces.backendMutex.RUnlock()
 
-		if cepCount < c.maxCEPsInCES && cepCount > largestCEPCount {
+		if ces.ces.Namespace == ns && cepCount < c.maxCEPsInCES && cepCount > largestCEPCount {
```

Comment:
I'd move namespace condition as a last one - it's quicker to compare integers than strings.

Reply:
Good suggestion. Done.